Quadriga/src/main/java/edu/asu/spring/quadriga/service/transformation/impl/TransformationManager.java

```diff
@@ -51,4 +55,68 @@ public void saveTransformations(ITransformationFile transformationFile) throws F
         return transformationDAO.getAllTransformations();
     }
 
+    /**
+     * Retrieve the transformation file object which contains the absolute file
+     * paths of patternFile and MappingFile .
+     * 
+     * @param transformationId
+     *            Transformation ID of the transformation.
+     * @return Returns the transformation file object
+     */
+    @Transactional
+    @Override
+    public ITransformationFile retrieveTransformationFilePaths(String transformationId) {
+
+        TransformFilesDTO transformDTO = (TransformFilesDTO) transformationDAO.getDTO(transformationId);
+
+        String fileLocation = transformationFileService.getTransformFileLocation();
+        String absolutePatternFilePath = getAbsoluteFilePath(fileLocation, transformationId,
+                transformDTO.getPatternFileName());
+        String absoluteMappingFilePath = getAbsoluteFilePath(fileLocation, transformationId,
+                transformDTO.getMappingFileName());
+
+        TransformationFile transformFile = new TransformationFile();
+        transformFile.setAbsolutePatternFilePath(absolutePatternFilePath);
+        transformFile.setAbsoluteMappingFilePath(absoluteMappingFilePath);
+
+        return transformFile;
+    }
+
+    /**
+     * appends the location with dirName and fileName and returns the resultant
+     * string as absoluteFilePath
+     * 
+     * @param location
+     * @param dirName
+     * @param fileName
+     * @return
+     */
+    private String getAbsoluteFilePath(String location, String dirName, String fileName) {
+
+        StringBuffer absoluteFilePath = new StringBuffer();
+
+        absoluteFilePath.append(location);
+        absoluteFilePath.append("/");
+        absoluteFilePath.append(dirName);
+        absoluteFilePath.append("/");
+        absoluteFilePath.append(fileName);
+
+        return absoluteFilePath.toString();
+    }
+
+    @Transactional
+    @Override
+    public List<ITransformation> getTransformations(String transformationId) {
```

Comment:
why is this called getTransformations if it only creates one ITransformation object?